iso15118/secc/controller/interface.py

```diff
@@ -689,6 +729,54 @@ async def get_dc_bpt_charge_params_v20(
         """
         raise NotImplementedError
 
+    @abstractmethod
+    async def get_scheduled_dc_charge_loop_params(
+        self,
+    ) -> ScheduledDCChargeLoopResParams:
+        """
+        Gets the parameters for the DCChargeLoopRes in the Scheduled control mode
+
+        Relevant for:
+        - ISO 15118-20
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_dynamic_dc_charge_loop_params(self) -> DynamicDCChargeLoopRes:
+        """
+        Gets the parameters for the DCChargeLoopRes in the Dynamic control mode
+
+        Relevant for:
+        - ISO 15118-20
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_bpt_scheduled_dc_charge_loop_params(
+        self,
+    ) -> BPTScheduledDCChargeLoopResParams:
+        """
+        Gets the parameters for the DCChargeLoopRes in the Scheduled control mode for
+        bi-directional power transfer (BPT)
+
+        Relevant for:
+        - ISO 15118-20
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_bpt_dynamic_dc_charge_loop_params(
+        self,
+    ) -> BPTDynamicDCChargeLoopRes:
+        """
+        Gets the parameters for the DCChargeLoopRes in the Dynamic control mode for
+        bi-directional power transfer (BPT)
```

Comment:
I dont think we need this level of segregation. We can get what we need with one call (named for example `get_evse_dc_limits` whose result shall encapsulate:
* EVSEMaxChargePower 
* EVSEMinChargePower
* EVSEMaxChargeCurrent
* EVSEMaxVoltage
* EVSEMinVoltage

If the EVSE does NOT support BPT, then the fields in the MQTT corresponding to the following ones can be set to 0. I would do that instead of allowing the values to be None, so that in case we are on a BPT session we always have values. 

* EVSEMaxDischargePower 
* EVSEMinDischargePower
* EVSEMaxDischargeCurrent


And actually, thinking about this, we should not request this values, neither the DepartureTime, neither the TargetSOC or MinSOC. All these values and the above ones shall be pushed, either by the manufacturer or by OCPP. Because, in the practical situation we may or may not have new targets and if it is the same why are we constantly sending a MQTT call and wasting time and cpu resources?

So, what we shall do is to request this call once: https://josev-api.switch-ev.com/external_messages/cs_status_and_limits

Then we need to define a new message, that is used to push the station limits/targets as mentioned here: https://www.notion.so/switchev/Load-Management-with-ISO-15118-2-and-wake-up-procedure-0cf256cf620e40119a184938ec43fc11#cbec90c7460f45ac995c741de0e22b42

I need to think on a good message name and structure to include all we need:
* schedule
* max/min power (active and reactive), current and voltage levels
* soc targets
* departure times

Reply:
Sure -  as discussed some of the methods have now been unified. This is a short summary of how the methods that have been updated:
```
get_ac_charge_params_v20()
get_ac_bpt_charge_params_v20()
==> get_ac_charge_params_v20(selected_service: ServiceV20)

get_dc_charge_params_v20()
get_dc_bpt_charge_params_v20()
==> get_dc_charge_params_v20(selected_service: ServiceV20)

get_scheduled_ac_charge_loop_params()
get_bpt_scheduled_ac_charge_loop_params()
get_dynamic_ac_charge_loop_params()
get_bpt_dynamic_ac_charge_loop_params()
==> get_ac_charge_loop_params_v20(control_mode, selected_service)

get_scheduled_dc_charge_loop_params()
get_bpt_scheduled_dc_charge_loop_params()
get_dynamic_dc_charge_loop_params()
get_bpt_dynamic_dc_charge_loop_params()
==> get_dc_charge_loop_params_v20(control_mode, selected_service)
```

Reply:
for historical reasons, I transcribe here the chat between I and Shalin:

> get_ac_charge_params_v20(selected_service: ServiceV20)
   get_dc_charge_params_v20(selected_service: ServiceV20)

those two we probably will get the values from the cs_status_and_limits
which we have to adapt to contain the max/min charge and discharge current/power values

I also realized something:
* In Dynamic DC Charge Loop (BPT or not) - The EVSE has full control over what the EV consumes at any point. This means that we dont need to change over time the EVSE limits provided to the EV. Those can be static since ChargeParameterDiscovery stage.
* In Scheduled DC Charge Loop (BPT or not) - The EVSE still has direct control over the power electronics. The difference here is that the EV is following a schedule, so it provides Targets to the EV based on the schedule. This is the equivalent DC use case of ISO 15118-2. In order to change the Current/Voltage requested by the EV we can either do a renegotiation or set the Max/Min limit of the station in the Res message
* In Dynamic AC Charge Loop (BPT or not) - The power electronics sits in the EV, so during this mode, the EV will tell the limits to the EVSE and the EVSE influences the EV with the Target Active and Reactive powers
* In Scheduled AC Charge Loop (BPT or not)  - The power electronics sits in the EV as well. This is the equivalent of the AC ISO15118-2 use case. The EV is just obligated to send the EVPresentActivePower and the EVSE can influence the schedule, either by renegotiation or by sending a Target


What is the interaction of Josev and the CS Hardware?
* In Dynamic DC Charge Loop (BPT or not): When Josev receives a new profile, it can directly send the new setpoint to the CS/power electronics of the DC station and no need to send anything to the ISO 15118
* In Scheduled DC Charge Loop (BPT or not): When Josev receives a new profile, it can directly send the new setpoint to the CS/power electronics of the DC station, but it may also need to change the Max/Min limits reported in ISO 15118 (if the setpoint deviates from the initial limits) or it can do a schedule renegotiation
* In Dynamic AC Charge Loop (BPT or not): When Josev receives a new profile, it  directly sends the new setpoint to the ISO 15118
* In Scheduled AC Charge Loop (BPT or not): When Josev receives a new profile, it  directly sends the new setpoint to the ISO 15118 or does a schedule renegotiation
When in basic charging [AC only]: when Josev has a new setpoint, it may use the cp_pwm message to change the duty cycle and influence the charging